### PR TITLE
[bugfix] fix broken vision inference, change serialization for bytes

### PR DESCRIPTION
**Pull request by yanxi0830**
# What does this PR do?

- vision inference via image as binary bytes fails with serialization error
- add custom serialization for "bytes" in `_URLOrData`

## Test Plan

```
pytest -v -s -k "fireworks" --inference-model="meta-llama/Llama-3.2-11B-Vision-Instruct" ./llama_stack/providers/tests/inference/test_vision_inference.py::TestVisionModelInference::test_vision_chat_completion_non_streaming
```

**Before**
image

**After**
image

image

## Sources

Please link relevant resources if necessary.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Ran pre-commit to handle lint / formatting issues.
- [ ] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [ ] Wrote necessary unit or integration tests.

### CODE REVIEW

File: llama_stack/apis/common/content_types.py [hunk 1]
```diff
@@ -27,6 +28,9 @@ def validator(cls, values):
             return values
         return {"url": values}
 
+    class Config:
```
Comment (dineshyv) [inline llama_stack/apis/common/content_types.py:31]: should we only do this if data, not URL?

Comment (yanxi0830) [reply llama_stack/apis/common/content_types.py:31]: URL should already be serializable (e.g. string1 test passes already).

File: llama_stack/distribution/configure.py [hunk 2]
```diff
@@ -176,6 +176,7 @@ def get_providers(entries):
     config_dict["apis"] = config_dict["apis_to_serve"]
     config_dict.pop("apis_to_serve", None)
 
+    print(config_dict)
```
Comment (dineshyv) [inline llama_stack/distribution/configure.py:179]: delete?

File: llama_stack/apis/common/content_types.py (hunk 1 above)
Comment (dineshyv) [reply llama_stack/apis/common/content_types.py:31]: where doe sit specify that its only for data and not url?

Comment (yanxi0830) [reply llama_stack/apis/common/content_types.py:31]: The key is a type that has custom serialization (e.g. here the key is "bytes"). 
```
json_encoders = {bytes: lambda v: base64.b64encode(v).decode("utf-8")}
```

Comment (yanxi0830) [reply llama_stack/apis/common/content_types.py:31]: actually just found out there's another way for custom serialization that specifically targets the `data` field.  https://docs.pydantic.dev/latest/concepts/serialization/#custom-serializers

**Review by dineshyv — APPROVED** [review]

**Pull request merged**